uamqp_generator/amqp_definitions_c.tt

```diff
@@ -194,71 +201,86 @@ int amqpvalue_get_<#= type_name #>(AMQP_VALUE value, <#= type_name.ToUpper() #>_
 <#					if (first_one) #>
 <#					{ #>
 <#						first_one = false; #>
-				AMQP_VALUE item_value;
+    				AMQP_VALUE item_value;
 <#					} #>
-				/* <#= field.name #> */
-				item_value = amqpvalue_get_list_item(list_value, <#= k #>);
-				if (item_value == NULL)
-				{
+				    /* <#= field.name #> */
+				    if (list_item_count > <#= k #>)
+                    {
+                        item_value = amqpvalue_get_list_item(list_value, <#= k #>);
+				        if (item_value == NULL)
+				        {
 <# 					if (field.mandatory == "true") #>
 <# 					{ #>
-					{
-						<#= type_name #>_destroy(*<#= type_name.ToLower() #>_handle);
-						result = __FAILURE__;
-						break;
-					}
+					        {
```

Comment:
Are these spaces or tabs? should follow the style from the rest of the document.